### PR TITLE
Preserve data in the browser

### PR DESCRIPTION
**Pull request by KAYEMBAJONATHAN**
- Add data.js on file.
- The data are saved in a local storage and preserved in the browser.

### CODE REVIEW

File: data.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+if (userInfo !== null) {
+  if (userInfo.hasOwnProperty.call('username')) {
+    nameInput.value = userInfo.username;
+  }
+  if (userInfo.hasOwnProperty.call('email')) {
+    emailInput.value = userInfo.email;
+  }
+  if (userInfo.hasOwnProperty.call('msg')) {
+    textAreaBoxInput.value = userInfo.msg;
+  }
```
Comment (leonardodiasb) [inline data.js]: - [ ] The data is being stored correctly to local storage, but after reloading the page, the `userInfo.hasOwnProperty.call('...')` is returning false even when there is data in the `userInfo` object, and this is preventing the page to load the data in the form. Kindly fix this bug and use another conditional to check if there is data in those data fields so they are correctly populated after the page reloads.

**Review by leonardodiasb — CHANGES_REQUESTED** [review]
# Required Changes ♻️ ⚠️ 🚦 

Hi @KAYEMBAJONATHAN, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

mario_ladders

## Good points 👍 

 - No linter errors. ✔️
 - Correct use of GitHub flow. ✔️ 

## Aspects to improve ♻️

- [x] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rloterh — COMMENTED** [review]
Hi Team,
 
 While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
 
 ## Invalid Code Review Request
You have submitted a project, whereby linter checks are not showing on your Pull Request. Kindly ensure that **all** required linters are properly configured with regard to GitHub actions, and **all** checks are passing. Kindly request another code review when this is done.
 
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
 Cheers and Happy coding!👏👏👏
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by medaminedev66** [comment]
Hello @KAYEMBAJONATHAN 👋,

_This is an additional review as a result of the second opinion request._

I can see that you have a conflict on your PR. That's why the Linters checks does not show up. I recommend solving the conflict by following the steps below:
- Try to identify the files that have conflicts in your pull request.
- Fetch the latest changes from the upstream branch to your local branch. This can be done using the following commands in your terminal:
```php
git fetch upstream
git checkout <branch_name>
git merge upstream/<branch_name>
```
- Once the latest changes are fetched, you need to resolve the conflict by editing the files with conflicts. The conflicts are marked in the file with <<<<<<<, =======, and >>>>>>> markers. You need to manually edit the file to remove the markers and keep the relevant changes.
- Once the conflict is resolved, you need to add the changes to the staging area and commit them with a commit message explaining the changes.
```sql
Copy code
git add .
git commit -m "Resolved conflict on <file_name>"
```
- Finally, you need to push the changes to your branch in the remote repository.
```perl
git push origin <branch_name>
```
- Rebase the branch: If your branch is behind the upstream branch, you may need to rebase your branch on top of the upstream branch. This can be done using the following command:
```php
git rebase upstream/<branch_name>
```
After completing these steps, you should be able to resolve the conflict on your pull request.

*If you still cannot solve the conflict, it's ok; no worries. You still can request a review and the next Code Reviewer will review your PR. But you still cannot merge your changes to the main branch if your PR is approved. That's why I highly recommend resolving the conflict.*

Keep rocking 🚀!

**Review by binhussen — APPROVED** [review]
Hi @sam2if and @KAYEMBAJONATHAN 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- All linter checks passed ✔️
- Descriptive pull request ✔️
- Used GitHub flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**